src/transformers/modeling_utils.py

```diff
@@ -689,21 +688,41 @@ def _find_identical(tensors: List[Set[str]], state_dict: Dict[str, torch.Tensor]
     return shared_tensors, identical
 
 
-def _load_state_dict_into_model(model_to_load, state_dict, start_prefix, assign_to_params_buffers=False):
+def _load_state_dict_into_model(
+    model_to_load, state_dict, start_prefix, assign_to_params_buffers=False, pretrained_model_name_or_path=None
+):
     # Convert old format to new format if needed from a PyTorch state_dict
     old_keys = []
     new_keys = []
+    renamed_keys = {}
+    found_gamma = False
+    found_beta = False
+    warning_msg = "This model "
+    if pretrained_model_name_or_path is not None:
+        warning_msg += f"(`{pretrained_model_name_or_path}`) "
+    warning_msg += "contains parameters that have been renamed internally (a few are listed below but more are present in the model):\n"
     for key in state_dict.keys():
         new_key = None
         if "gamma" in key:
-            logger.warning(PARAM_RENAME_WARNING.format("gamma", "weight"))
+            # We add only the first key as an example
+            if not found_gamma:
+                renamed_keys[key] = key.replace("gamma", "weight")
+            found_gamma = True
             new_key = key.replace("gamma", "weight")
```

Comment:
Note we want this to be and not or, because it's important that the user knows if both gamma or beta are present. Otherwise went with this solution!

Reply:
>  because it's important that the user knows if both gamma or beta are present.

Right, but I think in this case we'd still want to use `or`. If `and` is used, we only have when `key` contained both `gamma` and `beta`, but not if there were two separate values -- one key which contained `gamma` and one key which contained `beta`

Reply:
Correct, I misspoke what I actually wound up doing here. In the end it's `renamed_keys = {**renamed_gamma, **renamed_beta}` to catch everything